internal/discovery/discovery-config.go

```diff
@@ -138,17 +130,29 @@ func (discovery *DiscoveryCtrl) handleConfiguration(config configv1alpha1.Discov
 
 func (discovery *DiscoveryCtrl) reloadServer() {
 	klog.Info("Reload mDNS server")
-	select {
-	case discovery.stopMDNS <- true:
-		close(discovery.stopMDNS)
-	default:
-	}
+	discovery.stopMDNS <- true
 	if discovery.Config.EnableAdvertisement {
+		emptyChan(discovery.stopMDNS)
 		go discovery.Register()
 	}
 }
 
 func (discovery *DiscoveryCtrl) reloadClient() {
 	klog.Info("Reload mDNS client")
-	// settings are automatically updated in next iteration
+	discovery.stopMDNSClient <- true
+	if discovery.Config.EnableDiscovery {
+		emptyChan(discovery.stopMDNSClient)
+		go discovery.StartResolver(discovery.stopMDNSClient)
+	}
+}
+
+func emptyChan(ch <-chan bool) {
```

Comment:
This function is used to empty the two chans that control mdns client/server restart

Each time something that requires a client/server restart changes in ClusterConfig a new value is inserted in this chan. The problem is that if we have multiple updates in a short time or the component was off this can lead to multiple restarts of mdns

With this function, we can be sure that when we start the mdns components there are no old stuff in the control chans

Reply:
I think the this explanation can be useful to be used as comment.

Reply:
Couldn't we instantiate a new chan instead of using this approach? It looks very strange to me.

Reply:
Done